relay-server/src/actors/envelopes.rs

```diff
@@ -2616,6 +2684,53 @@ mod tests {
         assert_eq!(new_envelope.items().next().unwrap().ty(), ItemType::Event);
     }
 
+    #[test]
+    fn test_client_report_removal() {
+        relay_test::setup();
+
+        let processor = EnvelopeProcessor::new(Arc::new(Default::default()));
+
+        let dsn = "https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"
+            .parse()
+            .unwrap();
+
+        let request_meta = RequestMeta::new(dsn);
+        let mut envelope = Envelope::from_request(None, request_meta);
+
+        envelope.add_item({
+            let mut item = Item::new(ItemType::ClientReport);
+            item.set_payload(
+                ContentType::Json,
+                r###"
+                    {
+                    "discarded_events": [
+                        ["queue_full", "error", 42]
+                    ]
+                    }
+                "###,
+            );
+            item
+        });
+
+        let envelope_response = relay_test::with_system(move || {
```

Comment:
why this:

```
with_system(|| process(...).unwrap())
```

over this:

```
block_fn(|| process(...)).unwrap()
```

if block_fn silently discards errors we should fix that

Reply:
Because `block_fn` requires creating a future.

Reply:
block_fn requires intofuture which Result as returned by process implements